src/EditorFeatures/Core.Wpf/Adornments/AbstractAdornmentManager.cs

```diff
@@ -284,7 +284,7 @@ protected bool ShouldDrawTag(SnapshotSpan snapshotSpan, IMappingTagSpan<T> mappi
             }
 
             var mappedPoint = TextView.BufferGraph.MapUpToSnapshot(
-                point.Value, PointTrackingMode.Negative, PositionAffinity.Predecessor, TextView.VisualSnapshot);
+                point.Value, PointTrackingMode.Negative, PositionAffinity.Predecessor, TextView.TextSnapshot);
```

Comment:
Because I am curious, what is the difference between these?

Reply:
I am not quite sure what the difference is to be honest, but I did MapUpToSnapshot earlier in the file as well and I used the TextSnapshot. This was just to keep consistency because I don't think I used the VisualSnapshot for a particular reason.

Reply:
can you make sure that if you scroll these off screen that they disappear (debugging through to amek sure that happens would be good too).